### ISSUE TITLE
First Layer Height Can;t be Greater than Nozzle Diameter Error

### ISSUE THREAD

**Issue by AndyEveritt** [supermerill/SuperSlicer#1035]
### Version
2.3.55.5

### Operating system type + version
Windows 10 & Ubuntu 20.04

### 3D printer brand / version + firmware version (if known)
E3D Toolchanger. Reprapfirmware 3.2

#### Description
Get "First layer height can't be greater than nozzle diameter" error when using multi extruder printer with different nozzle diameters
![image](https://user-images.githubusercontent.com/38423143/112722459-9b68ad80-8f01-11eb-9c1e-e2b958b2ae53.png)

#### Steps to Reproduce
* Set the nozzle diameter for extruder 1 to be larger (eg 0.6 mm), and extruder 2 to be small (eg 0.3 mm)
  * ![image](https://user-images.githubusercontent.com/38423143/112722572-55f8b000-8f02-11eb-828a-d240bd46cbeb.png)
  * ![image](https://user-images.githubusercontent.com/38423143/112722577-601aae80-8f02-11eb-9936-1b61fd62c2ae.png)
* Set the first layer height to be a percentage
 * The value must be larger than `100 * (nozzle_diameter_2 / nozzle_diameter_1)%`
  * ![image](https://user-images.githubusercontent.com/38423143/112722631-9ce6a580-8f02-11eb-9e8b-5554b6996b1d.png)
* Change the extruder to be used for at least one of the parts on the build plate to extruder 2
* Try to slice

#### Expected Results
The first layer height percentage would be calculated from the nozzle diameter of the extruder that is being used for the specific part.

#### Actual Results
The first layer height percentage is calculated from the nozzle diameter of extruder 1 even if it is not the one being used.

#### Project File (.3MF) where problem occurs

[first layer height error.zip](https://github.com/supermerill/SuperSlicer/files/6215758/first.layer.height.error.zip)

**Comment by supermerill** [comment]
yes, it's an unfortunate problem.
in the tooltip I wrote "from default extruder" but I should have written, "from the first extruder".

Problem: I have a collection of extruders used in a print.
what is the one I should take? the bigger one

**Comment by AndyEveritt** [comment]
Yeh the tooltip does line up with what is happening but I think what is happening should be changed.

The easiest fix to remove the error from occuring would be to calculate the layer height from the smallest nozzle diameter. However I think the better solution, that would make it more consistent with layer width percentage calculation, would be to calculate each objects first layer height independently.

**Comment by AndyEveritt** [comment]
I am not familiar with the technicalities of how PS/SS deals with various layer heigths for different objects but it must support it to some extent due to the variable layer height tool

**Comment by AndyEveritt** [comment]
I don't think you can set multiple extruders for a single object in PS but may be wrong 
![image](https://user-images.githubusercontent.com/38423143/113157227-9c1b7f80-9232-11eb-83a3-04a6cde8a023.png)

If you can, then yes, I think take the smallest.

**Comment by supermerill** [comment]
You can, by using a dirrefent extruder for perimeter, infil... also there is the support extruder that is used for the whole plater.
You can also add parts into an object with a different extruder.

**Comment by supermerill** [comment]
I'll see if I can grab the information everywhere it's needed.

**Comment by AndyEveritt** [comment]
That's a very good point. Using all nozzle diameters used in each object seems to be making the most sense in that case
